module_message.lua

```diff
@@ -462,9 +483,14 @@ function Module:OnMessageCreate(message)
 	local config = self:GetConfig(message.guild)
 	local reply = config.Replies[message.content]
 	if (reply) then
-		local success, err = message:reply(reply)
+		reply = ShallowCopy(reply)
+		local deleteInvokation = RemoveTableKey(reply, "deleteInvokation")
+	 	local success, err = message:reply(reply)
+		
 		if (not success) then
```

Comment:
It would be better to make it like:
```lua
if (success) then
  -- Handle delete invokation
else
-- Log error
end
```

Reply:
```lua
if (not success) then
  self:LogError(message.guild, "Failed to reply to %s: %s", message.content, err)
elseif (deleteInvokation == nil or deleteInvokation and config.DeleteInvokation) then
  message:delete()
end
```
imo, this is better than
```lua
if (success) then
  if (deleteInvokation == nil or deleteInvokation and config.DeleteInvokation) then
    message:delete()
  end
else
  self:LogError(message.guild, "Failed to reply to %s: %s", message.content, err)
end
```

Reply:
I disagree, I find the elseif pretty hard to understand here.
Maybe
```lua
if (not success) then
    self:LogError(message.guild, "Failed to reply to %s: %s", message.content, err)
    return
end

-- handle delete invokation
```
?